exporter/sumologicexporter/prometheus_formatter.go

```diff
@@ -0,0 +1,445 @@
+// for each data points: one per bound (with one for infinity) plus two for sum and count
+func (f *prometheusFormatter) intHistogram2Strings(record metricPair) []string {
+	dps := record.metric.IntHistogram().DataPoints()
+	var lines []string
+
+	for i := 0; i < dps.Len(); i++ {
+		dp := dps.At(i)
+
+		explicitBounds := dp.ExplicitBounds()
+		var cumulative uint64
+		additionalAttributes := pdata.NewAttributeMap()
+
+		for i, bound := range explicitBounds {
+			cumulative += dp.BucketCounts()[i]
+			additionalAttributes.UpsertDouble("le", bound)
+
+			line := f.uintValueLine(
+				record.metric.Name(),
+				cumulative,
+				dp,
+				f.mergeAttributes(record.attributes, additionalAttributes),
+			)
+			lines = append(lines, line)
+		}
+
+		cumulative += dp.BucketCounts()[len(explicitBounds)]
+		additionalAttributes.UpsertString("le", "+Inf")
+		line := f.uintValueLine(
+			record.metric.Name(),
+			cumulative,
+			dp,
+			f.mergeAttributes(record.attributes, additionalAttributes),
+		)
+		lines = append(lines, line)
+
+		line = f.intValueLine(
+			f.sumMetric(record.metric.Name()),
+			dp.Sum(),
+			dp,
+			record.attributes,
+		)
+		lines = append(lines, line)
+
+		line = f.uintValueLine(
+			f.countMetric(record.metric.Name()),
+			dp.Count(),
+			dp,
+			record.attributes,
+		)
+		lines = append(lines, line)
+	}
+
+	return lines
+}
+
+// doubleHistogram2Strings converts DoubleHistogram record to list of strings
+// for each data points: one per bound (with one for infinity) plus two for sum and count
+func (f *prometheusFormatter) doubleHistogram2Strings(record metricPair) []string {
+	dps := record.metric.DoubleHistogram().DataPoints()
+	var lines []string
+
+	for i := 0; i < dps.Len(); i++ {
+		dp := dps.At(i)
+
+		explicitBounds := dp.ExplicitBounds()
+		var cumulative uint64
+		additionalAttributes := pdata.NewAttributeMap()
+
+		for i, bound := range explicitBounds {
+			cumulative += dp.BucketCounts()[i]
+			additionalAttributes.UpsertDouble("le", bound)
+
+			line := f.uintValueLine(
+				record.metric.Name(),
+				cumulative,
+				dp,
+				f.mergeAttributes(record.attributes, additionalAttributes),
+			)
+			lines = append(lines, line)
+		}
+
+		cumulative += dp.BucketCounts()[len(explicitBounds)]
+		additionalAttributes.UpsertString("le", "+Inf")
+		line := f.uintValueLine(
+			record.metric.Name(),
+			cumulative,
+			dp,
+			f.mergeAttributes(record.attributes, additionalAttributes),
+		)
+		lines = append(lines, line)
+
+		line = f.doubleValueLine(
+			f.sumMetric(record.metric.Name()),
+			dp.Sum(),
+			dp,
+			record.attributes,
+		)
+		lines = append(lines, line)
+
+		line = f.uintValueLine(
+			f.countMetric(record.metric.Name()),
+			dp.Count(),
+			dp,
+			record.attributes,
+		)
+		lines = append(lines, line)
+	}
+
+	return lines
+}
+
+// metric2String returns stringified metricPair
+func (f *prometheusFormatter) metric2String(record metricPair) (string, error) {
+	var nextLines []string
```

Comment:
`nextLines` seems to have a meaning which isn't needed here? Not sure. This could simply be something like `str` or something similarly meaningless. It's just a return value.